irr/src/CIrrDeviceSDL.cpp

```diff
@@ -708,20 +819,46 @@ bool CIrrDeviceSDL::run()
 	SEvent irrevent;
 	SDL_Event SDL_event;
 
+	auto get_touch_id_x_y = [this, &irrevent, SDL_event]() {
```

Comment:
`SDL_event` is not set when this lambda is created so

```suggestion
	auto get_touch_id_x_y = [this, &irrevent, &SDL_event]() {
```

Reply:
Apparently this is a roulette (especially when you re-run cmake) that potentially breaks touch input. I tried it yesterday and touch input didn't work; when I tested https://github.com/luanti-org/luanti/pull/16583#discussion_r2529840032 (before you addressed it) the optimization was apparently applied (-> touch input worked). Then I pulled again (with debugging) and touch input was broken again.

Reply:
Perhaps. It does not matter. I'll keep it like this to be explicit about what's needed and what's not touched by the function.